novoda-blog-diversity/lib/data-source.js

```diff
@@ -19,97 +19,49 @@ const parseRss = (url) => {
 }
 
 const toViewState = (items) => {
-    var mentions = new Map([["android", 0], ["ios", 0], ["cross-platform", 0]])
-    mentions.set("android", counUniqueTagsIn(items, androidTags))
-    mentions.set("ios", counUniqueTagsIn(items, iosTags))
-    mentions.set("cross-platform", counUniqueTagsIn(items, crossPlatformTags))
-        
+    var map = tagsWithUsagesFrom(items)
     return {
-        mostUsedTags: mostUsedTagsFrom(items),
-        crossPlatformMentions: mentions.get("cross-platform"),
-        androidMentions: mentions.get("android"),
-        iosMentions: mentions.get("ios")
+        mostUsedTags: mostUsedTagsFrom(map),
+        crossPlatformMentions: counUniqueTagsIn(map, crossPlatformTags),
+        androidMentions: counUniqueTagsIn(map, androidTags),
+        iosMentions: counUniqueTagsIn(map, iosTags)
     }
 }
 
-function mostUsedTagsFrom(items){
-    return sortMapDescendingByValue(tagsWithUsagesFrom(items))
-        .splice(0,3)
-        .map(tagUsage => tagUsage[0])
+function mostUsedTagsFrom(map) {
+    return Object.keys(map)
+        .map(key => [key, map[key]]) // transforms map into array
+        .sort((firstTag, secondTag) => secondTag[1] - firstTag[1]) // sort array by value
+        .splice(0, 3) // take the first 3 elements
+        .map(tagUsage => tagUsage[0]) // take only the names
         .join(", ")
 }
 
-function tagsWithUsagesFrom(items){
+function tagsWithUsagesFrom(items) {
```

Comment:
I'm still not happy with this function. It must be possible to do this without creating a `Map`, because if you check later the map is always transformed to an array to be consumed